### PR TITLE
control-service: refactor service user doc

### PR DESCRIPTION
**Pull request by mrMoZ1**
what: updated service user setup documentation

why: it was(is) potentially unclear

testing: n/a doc change.

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties [hunk 1]
```diff
@@ -231,18 +231,27 @@ datajobs.aws.secretAccessKey=${AWS_ACCESS_KEY_SECRET}
...
+# By using a separate IAM user, the principle of least privilege is enforced.
 # https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
-# These need to be stored in different variables than the AWS credentials - datajobs.aws.accessKeyId
-# and secretAccessKey used by the control service, because if the K8S cluster is hosted on a EKS
-# instance, some API calls to the instance made by the control-service could fail.
+# The service user variables are different than the AWS credentials - datajobs.aws.accessKeyId
+# and secretAccessKey used by the control service.
+#
+# Necessary steps to setup Service Account:
+# - Create a Service Account in Amazon (regular account that will be granted permissions to ECR repo)
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:244]: Do you mean https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html#id_users_create_console ?

Comment (mrMoZ1) [reply projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:244]: yes

**Review by murphp15 — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties [hunk 2]
```diff
@@ -231,18 +231,28 @@ datajobs.aws.secretAccessKey=${AWS_ACCESS_KEY_SECRET}
...
-# and secretAccessKey used by the control service, because if the K8S cluster is hosted on a EKS
-# instance, some API calls to the instance made by the control-service could fail.
+# The service user variables are different than the AWS credentials - datajobs.aws.accessKeyId
+# and secretAccessKey used by the control service.
+#
+# Necessary steps to setup Service Account:
+# - Create a Service Account in Amazon (regular account that will be granted permissions to ECR repo) e.g:
+#   https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html#id_users_create_console
+#   e.g - IAM user: svc.example-service-user
+# - Create a role in your AWS account (account where your ECR repository lives) - https://docs.aws.amazon.com/cli/latest/reference/iam/create-role.html
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:247]: When creating a role I must provide `assume-role-policy-document` What should this look like ?

File: projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties [hunk 3]
```diff
@@ -231,18 +231,28 @@ datajobs.aws.secretAccessKey=${AWS_ACCESS_KEY_SECRET}
...
+#   e.g - IAMRole: svc.example-service-user, the role must have the same name as your service user.
+# - Create a trust relationship on the role to the control account - edit your newly created role's Trust Relationship:
+#   https://aws.amazon.com/blogs/security/how-to-use-external-id-when-granting-access-to-your-aws-resources/
+#   https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition.html.
+# - Verify access to the linked account:
+#   aws sts get-caller-identity with the Service Account credentials
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:253]: How do I obtain the service account credentials ?

File: projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties [hunk 4]
```diff
@@ -231,18 +231,28 @@ datajobs.aws.secretAccessKey=${AWS_ACCESS_KEY_SECRET}
...
+#
+# Necessary steps to setup Service Account:
+# - Create a Service Account in Amazon (regular account that will be granted permissions to ECR repo) e.g:
+#   https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html#id_users_create_console
+#   e.g - IAM user: svc.example-service-user
+# - Create a role in your AWS account (account where your ECR repository lives) - https://docs.aws.amazon.com/cli/latest/reference/iam/create-role.html
+#   e.g - IAMRole: svc.example-service-user, the role must have the same name as your service user.
+# - Create a trust relationship on the role to the control account - edit your newly created role's Trust Relationship:
+#   https://aws.amazon.com/blogs/security/how-to-use-external-id-when-granting-access-to-your-aws-resources/
+#   https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition.html.
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:251]: What needs to be edited ? 

I am trying to follow the instructions but I am a bit stucked here.

**Review by tozka — CHANGES_REQUESTED** [review]

**Review by antoniivanov — APPROVED** [review]
Let's merge this  It has useful information even without addressing the feedback.

**Pull request merged**